lib/auth/machineid/machineidv1/workload_identity_service_test.go

```diff
@@ -0,0 +1,226 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package machineidv1_test
+
+import (
+	"context"
+	"crypto/x509"
+	"testing"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/durationpb"
+
+	machineidv1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/machineid/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/auth/machineid/machineidv1/experiment"
+	"github.com/gravitational/teleport/lib/auth/native"
+)
+
+// TestWorkloadIdentityService_SignX509SVIDs is an integration test that uses a
+// real gRPC client/server.
+func TestWorkloadIdentityService_SignX509SVIDs(t *testing.T) {
```

Comment:
For all the code, do we have only one test? I don't think it covers all added logic.

Reply:
There's are more unit-tests tests introduced in https://github.com/gravitational/teleport/pull/38174 / https://github.com/gravitational/teleport/pull/38172 - however - for this PR it made the most sense to me to have a higher level integration test which assures the behaviour against the specification. This test will be far more resilient to changes and isn't reliant on the implementation details of this code.